emission/tests/pipelineTests/TestExpectationNotificationConfig.py

```diff
@@ -0,0 +1,206 @@
+
+
+    def testModeMatchesDate(self):
+        mode_intensive = eace._config["modes"][0]
+        mode_relaxed = eace._config["modes"][1]
+
+        # Case 1: no schedule
+        for dt in self.test_dates.values():
+            self.assertTrue(eace._mode_matches_date(mode_relaxed, dt))
+
+        # Case 2: yes schedule and it matches
+        for testlabel in {"first_intensive", "nth_intensive", "mth_intensive"}:
+            self.assertTrue(eace._mode_matches_date(mode_intensive, self.test_dates[testlabel]))
+        
+        # Case 3: yes schedule and it doesn't match
+        for testlabel in {"before_intensive", "first_relaxed", "mth_relaxed"}:
+            self.assertFalse(eace._mode_matches_date(mode_intensive, self.test_dates[testlabel]))
+
+
+    def testGetCollectionModeBySchedule(self):
+        answers = {
+            "before_intensive": "relaxed",
+            "first_intensive":  "intensive",
+            "first_relaxed":    "relaxed",
+            "nth_intensive":    "intensive",
+            "mth_intensive":    "intensive",
+            "mth_relaxed":      "relaxed"
+        }
+
+        for label,trip in self.fake_trips.items():
+            self.assertEqual(eace._get_collection_mode_by_schedule(trip)["label"], answers[label])
+
+
+    def testRuleMatchesLabel(self):
+        # An answers entry is a (labels, list of indices of matching rules in intensive mode, same for relaxed mode) tuple
+        answers = [
+            ([{"labels": {"label2": "value8"}, "p": 1}],    [0],   [0]),  # label1 is not in here, so it should match the red rule when we test with label1 below
+            ([{"labels": {"label1": "value1"}, "p": 1}],    [1],   [1]),
+            ([{"labels": {"label1": "value2"}, "p": 0.96}], [1],   [1]),
+            ([{"labels": {"label1": "value3"}, "p": 0.95}], [1],   [1,2]),
+            ([{"labels": {"label1": "value4"}, "p": 0.75}], [1,2], [1,2,3]),
+            ([{"labels": {"label1": "value5"}, "p": 0.65}], [0],   [1,2,3]),
+            ([{"labels": {"label1": "value6"}, "p": 0.55}], [0],   [0]),
+            ([{"labels": {"label1": "value7"}, "p": 0}],    [0],   [0])
+        ]
+
+        for labels,intensive_answer,relaxed_answer in answers:
+            for i_mode,answer in [(0, intensive_answer), (1, relaxed_answer)]:
+                mode = eace._config["modes"][i_mode]
+                for i_rule,rule in enumerate(mode["rules"]):
+                    rule = mode["rules"][i_rule]
+                    trip = {"data": {"inferred_labels": labels}}
+                    guess = eace._rule_matches_label(mode, rule, trip, "label1")
+                    if i_rule in answer: self.assertTrue(guess)
+                    else: self.assertFalse(guess)
+
+
+    def testGetRuleForLabel(self):
+        # An answers entry is a (labels, index of corresponding rule under intensive mode, index of corresponding rule under relaxed mode) tuple
```

Comment:
nit: expand on this a bit to explain how you go from the list of all rules to the selected rule.

Reply:
Please handle this in a subsequent commit.

Reply:
I'm not sure I understand what you mean.